src/deadbot.py

```diff
@@ -37,6 +40,71 @@ def _update_page(self, page):
             print(f"No changes made to '{page.name}'")
 
 
-if __name__ == '__main__':
-    bot = DeadBot()
-    bot.run()
+def parse_arguments():
+    parser = argparse.ArgumentParser(
+        prog="DeadBot",
+        description="Bot that lives to serve deadlocked.wiki",
+        epilog="Process Deadlock game files and extract data and stats",
+    )
+
+    # Setup / Base config Flags
+    parser.add_argument(
+        "-i",
+        "--dl_path",
+        help="Path to Deadlock game files (also set with DEADLOCK_PATH environment variable)",
+        default=os.getenv("DEADLOCK_PATH"),
+    )
+    parser.add_argument(
+        "-w",
+        "--workdir",
+        help="Directory for temp working files (also set with WORK_DIR environment variable)",
+        default=os.getenv(
+            "WORK_DIR", os.path.abspath(os.getcwd()) + "/decompiled-data"
+        ),
+    )
+    parser.add_argument(
+        "-o",
+        "--output",
+        help="Output directory (also set with OUTPUT_DIR environment variable)",
+        default=os.getenv("OUTPUT_DIR", os.path.abspath(os.getcwd()) + "/output-data"),
+    )
+    parser.add_argument(
+        "--decompiler_cmd",
+        help="Command for Valve Resource Format tool (also set with DECOMPILER_CMD env variable)",
+        default=os.getenv("DECOMPILER_CMD", "tools/Decompiler"),
+    )
+
+    # Operational Flags
+    parser.add_argument(
+        "-d",
+        "--decompile",
+        action="store_true",
+        help="Decompiles Deadlock game files. (also set with DECOMPILE environment variable)",
+    )
+    parser.add_argument(
+        "-b",
+        "--bot_push",
+        action="store_true",
+        help="Push current data to wiki (also set with BOT_PUSH environment variable)",
```

Comment:
can this default to `BOT_PUSH` env var?

Reply:
Yep!  It will default to that var when it checks for the parameter and it didn't "store_true".
If the env var isn't set, it'll default to false